exovetter/vetters.py

```diff
@@ -132,16 +119,122 @@ def run(self, tce, lightcurve):
             'norm_lpp': self.norm_lpp,
             'plot_data': self.plot_data}
 
-    def plot(self):
+    def plot(self):  # pragma: no cover
         if self.plot_data is None:
             raise ValueError(
                 'LPP plot data is empty. Execute self.run(...) first.')
 
         # target is populated in TCE, assume it already exists.
-        target = self.tce['target_name']
+        target = self.tce.get('target_name', 'Target')
         lpp.plot_lpp_diagnostic(self.plot_data, target, self.norm_lpp)
 
 
+class Sweet(BaseVetter):
+    """Class to handle SWEET Vetter functionality.
+
+    Parameters
+    ----------
+    threshold_sigma : float
+        Threshold for comparing signal to transit period.
+
+    Attributes
+    ----------
+    tce, lc
+        Inputs to :meth:`run`.
+
+    result : dict
+        ``'amp'`` contains the best fit amplitude, its uncertainty, and
+        amplitude-to-uncertainty ratio for half-period, period, and
+        twice the period. ``'msg'`` contains warnings, if applicable.
+        Populated by :meth:`run`.
+
+    lsf : `~exovetter.utils.WqedLSF`
+        Least squares fit object, populated by :meth:`run`.
+
+    """
+    def __init__(self, threshold_sigma=3):
+        self.tce = None
+        self.lc = None
+        self.result = None
+        self.lsf = None
+        self.threshold_sigma = threshold_sigma
+
+    def run(self, tce, lightcurve):
+        self.tce = tce
+        self.lc = lightcurve
+        epoch = tce.get_epoch(getattr(exo_const, lightcurve.time_format))
+
+        # TODO: Do we want results to have unit? If so, what?
+        self.result, self.lsf = self._do_fit(
+            lightcurve.time * u.day, lightcurve.flux_quantity,
+            tce['period'], epoch, tce['duration'])
+
+    def plot(self):  # pragma: no cover
+        import matplotlib.pyplot as plt
+
+        phase = self.lsf.x
+        flux = self.lsf.y
+        best_fit = self.lsf.get_best_fit_model()
+
+        fig, axes = plt.subplots(2, 1, sharex=True)
+        plt.subplots_adjust(hspace=0.001)
+        axes[0].plot(phase, flux, 'k.')
+        axes[1].plot(phase, best_fit, 'r.')
+        axes[0].set_ylabel('Flux')
+        axes[1].set_xlabel('Phase')
+        axes[0].set_title(f'{self.tce.get("target_name", "Target")} '
+                          f'({self.tce.get("event_name", "Event")})')
+        plt.draw()
+
+        return fig
+
+    def _do_fit(self, time, flux, period_in, epoch, duration_in):
```

Comment:
I think it makes sense to break off _do_fit into a separate, sweet.py module. That better separates the interface from the implementation.

Reply:
Sure, in that case, might as well rename `utils.py` to `sweet.py`. Yes, I can do that.

Reply:
Make a separate sweet.py. There are some functions in utils.py that will be generally useful (e.g mark_transit_cadences)

Reply:
Sure. I'll do that along with the other changes once I hear back from you whether to subtract mean or not. Thanks!